### PR TITLE
feat: support HTTP2 for hertz client and server

### PR DESCRIPTION
**Pull request by Duslia**
#### What type of PR is this?

feat
#### What this PR does / why we need it (English/Chinese):

en: support HTTP2 for hertz client and server
zh: hertz client 和 server 支持 HTTP2

### CODE REVIEW

File: go111.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2018 The Go Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file.
+ */
+
+package http2
```
Comment (efectn) [inline go111.go:24]: I think we don't need go111.go & not_go111.go

**Review by efectn — COMMENTED** [review]
You can add https://github.com/summerwind/h2spec for testing. It has testing conditions for http2 so much.

**Comment by Duslia** [comment]
> You can add https://github.com/summerwind/h2spec for testing. It has testing conditions for http2 so much.

Ok, I will have a try. Thanks a lot.

**Comment by Duslia** [comment]
I use this repo `https://github.com/summerwind/h2spec` to run h2spec test. And it get the same result with go net. 
image
These are all expected. For more information, please refer to https://github.com/golang/go/issues/56760

**Comment by li-jin-gou** [comment]
All other projects have a main branch, do we need to be consistent here?

**Comment by Duslia** [comment]
> All other projects have a main branch, do we need to be consistent here?

done

**Review by li-jin-gou — APPROVED** [review]

**Review by welkeyever — APPROVED** [review]

**Pull request merged**